kubectl-fdb/cmd/k8s_client.go

```diff
@@ -184,18 +185,24 @@ func getAllPodsFromClusterWithCondition(kubeClient client.Client, clusterName st
 		return podNames, err
 	}
 
+	podMap := make(map[string]corev1.Pod)
 	for _, pod := range pods.Items {
-		for process := range processesSet {
-			if pod.Status.Phase != corev1.PodRunning {
-				continue
-			}
-
-			if pod.Labels[cluster.GetProcessGroupIDLabel()] != string(process) {
-				continue
-			}
+		podMap[pod.Labels[cluster.GetProcessGroupIDLabel()]] = pod
+	}
 
-			podNames = append(podNames, pod.Name)
+	for process := range processesSet {
+		if _, ok := podMap[string(process)]; !ok {
+			log.Printf("Pod for process group %s not found", process)
```

Comment:
Can we pass down the error o writer from the command line? Otherwise we will print this to stdout (always), which makes it harder to test and the warning should also be on stderr.